### ISSUE TITLE
Menu items are not in italic, sections is.

### ISSUE THREAD

**Issue by kierun** [squidfunk/mkdocs-material#3293]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

![screenshot](https://user-images.githubusercontent.com/3079575/144606956-0f985fc0-30a1-4d64-b91e-46bcb577b19a.png)

The menu showing the different sections does not support italic. However, the section does.

### Expected behaviour

The menu should be in italic as well.

### Actual behaviour

The menu is not in italic.

### Steps to reproduce

1. Create a markdown file with an italic in the title: `# ook _eek_ ook`
2. Run `mkdocs serve`
3. Look at the page, especially at the menu.

### Package versions

- Python: `python --version`
- MkDocs: `mkdocs --version`
- Material: `pip show mkdocs-material | grep -E ^Version`

```bash
√ ; python --version
Python 3.10.0
√ ; mkdocs --version
mkdocs, version 1.2.3 from /home/yann/.pyenv/versions/3.10.0/lib/python3.10/site-packages/mkdocs (Python 3.10)
√ ; pip show mkdocs-material | grep -E "^Version"
Version: 8.0.2
```

### Configuration

```yaml
site_name: My Top Secret stuff™
theme:
  name: material
  font:
    text: "PT Sans Narrow"
  favicon: assets/favicon.png
  palette:
    scheme: slate
    primary: teal
  logo: assets/logo.png
  features:
    - navigation.instant
markdown_extensions:
  - admonition
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:materialx.emoji.to_svg
  - pymdownx.highlight:
      use_pygments: true
      auto_title: true
      linenums: true
      linenums_style: pymdownx-inline
  - pymdownx.superfences
```

### System information

- Operating system: ...
- Browser: ...
```bash
√ ; uname -a
Linux nightwatch.neverness.org 5.15.6-100.fc34.x86_64 #1 SMP Wed Dec 1 13:41:51 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
√ ; cat /etc/redhat-release
───────┬────────────────────────────────────────────────────────────────────────
       │ File: /etc/redhat-release
───────┼────────────────────────────────────────────────────────────────────────
   1   │ Fedora release 34 (Thirty Four)
───────┴────────────────────────────────────────────────────────────────────────
√ ; firefox --versino
√ ; firefox --version
Mozilla Firefox 94.0
```

**Comment by squidfunk** [comment]
Thanks for reporting. This is not a problem of Material for MkDocs, but of MkDocs or Python Markdown, since the theme seems to get the raw Markdown (with the `_..._` modifiers). Please report this upstream.

**Comment by kierun** [comment]
Ah, my apologies for the noise. ☹

**Comment by kierun** [comment]
@squidfunk I am so “stealing” your bug report configuration! It's ace. :grin:

**Comment by squidfunk** [comment]
I'm happy to announce that this is now finally possible with the [built-in typeset plugin](https://squidfunk.github.io/mkdocs-material/reference/#built-in-typeset-plugin) that was just released as part of [Insiders 4.27.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.27.0)! With the plugin active, the navigation and table of contents sidebars will render headlines with HTML.
